pkgs/racket-doc/scribblings/reference/trace.scrbl

```diff
@@ -165,6 +175,104 @@ traced call. It receives the name of the function, the function's
 number indicating the depth of the call.
 
+Modifying this is useful to get more readable or additional output when tracing syntax transformers.
+For example, we may only care about the shape of the syntax object and not the source location
+information.
+By modifying @racket[current-trace-print-args], we can print the argument to (and, using
+@racket[current-trace-print-result], the result of) the syntax transformer as a datum using
+@racket[syntax->datum], as follows.
+
+@examples[#:eval ev
+  (require (for-syntax racket/trace))
+  (begin-for-syntax
+    (current-trace-print-args
+      (let ([ctpa (current-trace-print-args)])
+        (lambda (s l kw l2 n)
+          (ctpa s (map syntax->datum l) kw l2 n))))
+    (current-trace-print-results
+      (let ([ctpr (current-trace-print-results)])
+        (lambda (s r n)
+         (ctpr s (map syntax->datum r) n)))))
+
+  (trace-define-syntax fact
+    (syntax-rules ()
+      [(_ x) 120]))
+  (fact 5)]
+
+In this example, we update the printers by storing the current printers, @racket[ctpa] and
+@racket[ctpr], which we call after converting the arguments @racket[l] and results @racket[r] to
+datum.
+
+Note that this modification of @racket[current-trace-print-args] and
+@racket[current-trace-print-result] is an imperative update, and will affect all traced identifiers.
+This example assumes all arguments and results to @emph{all traced functions} will be syntax objects,
+which is the case only if you are only tracing syntax transformers.
+If used as-is, the above code could result in type errors when tracing both functions and syntax transformers.
+It would be better to use @racket[syntax->datum] only when the argument or result is actually a syntax
+object, for example, by defining @racket[maybe-syntax->datum] as follows.
+
+@examples[#:eval ev1
+  (require (for-syntax racket/trace))
+  (begin-for-syntax
+    (define (maybe-syntax->datum syn?)
+      (if (syntax? syn?)
+          (syntax->datum syn?)
+          syn?))
+    (current-trace-print-args
+      (let ([ctpa (current-trace-print-args)])
+        (lambda (s l kw l2 n)
+          (ctpa s (map maybe-syntax->datum l) kw l2 n))))
+    (current-trace-print-results
+      (let ([ctpr (current-trace-print-results)])
+        (lambda (s l n)
+         (ctpr s (map maybe-syntax->datum l) n))))
+
+  (trace-define (precompute-fact syn n) (datum->syntax syn (apply * (build-list n add1)))))
+  (trace-define (run-time-fact n) (apply * (build-list n add1)))
+
+  (require (for-syntax syntax/parse))
+  (trace-define-syntax (fact syn)
+    (syntax-parse syn
+      [(_ x:nat) (precompute-fact syn (syntax->datum #'x))]
+      [(_ x) #'(run-time-fact x)]))
+  (fact 5)
+  (fact (+ 2 3))]}
+
+Finally, you can modify these parameters to print additional information about syntax objects.
+For example, we can use @racketmodname[debug-scopes] to trace how scopes change through syntax
+transformers.
+
+@; manually typeset to avoid depending on debug-scopes
```

Comment:
Hmm, I think it would be better to add this example to the `debug-scopes` docs and just point to it from here --- so the example doesn't need to be manually typeset

Reply:
That's not a bad idea. I'll see if I can get this snippet merged into that package first, then update this PR.